src/dmd/semantic3.d

```diff
@@ -875,7 +875,6 @@ extern(C++) final class Semantic3Visitor : Visitor
                 sc2.flags = (sc2.flags & ~SCOPEcontract) | SCOPErequire;
 
                 // BUG: need to error if accessing out parameters
-                // BUG: need to treat parameters as const
```

Comment:
Hmm. This PR seems to only disable writing to `this`, but does it also prohibit modifying function parameters?

Reply:
@quickfur that was implemented in #1569.  Look at the _whole_ of checkModify.

Reply:
Ah, ok. Nevermind then. :-)   I was just confused because deleting this comment in this PR makes it seem like this PR is fixing the entire issue. I suppose making `this` const is the last remaining issue as far as this comment is concerned?